tests/integ/scala/test_datatype_suite.py

```diff
@@ -970,3 +970,101 @@ def test_structured_type_print_schema(
+        pytest.skip("Test requires structured type support.")
+
+    table_name = f"snowpark_schema_expresion_test_{uuid.uuid4().hex[:5]}".upper()
+    non_null_table_name = (
+        f"snowpark_schema_expresion_nonnull_test_{uuid.uuid4().hex[:5]}".upper()
+    )
+
+    expected_schema = StructType(
+        [
+            StructField(
+                "MAP",
+                MapType(StringType(), DoubleType(), structured=True),
+                nullable=True,
+            ),
+            StructField("ARR", ArrayType(DoubleType(), structured=True), nullable=True),
+            StructField(
+                "OBJ",
+                StructType(
+                    [
+                        StructField("FIELD1", StringType(), nullable=True),
+                        StructField("FIELD2", DoubleType(), nullable=True),
+                    ],
+                    structured=True,
+                ),
+                nullable=True,
+            ),
+        ]
+    )
+
+    expected_non_null_schema = StructType(
+        [
+            StructField(
+                "MAP",
+                MapType(StringType(), DoubleType(), structured=True),
+                nullable=False,
+            ),
+            StructField(
+                "ARR", ArrayType(DoubleType(), structured=True), nullable=False
+            ),
+            StructField(
+                "OBJ",
+                StructType(
+                    [
+                        StructField("FIELD1", StringType(), nullable=False),
+                        StructField("FIELD2", DoubleType(), nullable=False),
+                    ],
+                    structured=True,
+                ),
+                nullable=False,
+            ),
+        ]
+    )
+    try:
+        # SNOW-1819428: Nullability doesn't seem to be respected when creating
+        # a structured type dataframe so use a table instead.
+        structured_type_session.sql(
+            f"create table {table_name} (MAP MAP(VARCHAR, DOUBLE), ARR ARRAY(DOUBLE), "
+            "OBJ OBJECT(FIELD1 VARCHAR, FIELD2 DOUBLE))"
+        ).collect()
+        structured_type_session.sql(
+            f"create table {non_null_table_name} (MAP MAP(VARCHAR, DOUBLE) NOT NULL, "
+            "ARR ARRAY(DOUBLE) NOT NULL, OBJ OBJECT(FIELD1 VARCHAR NOT NULL, FIELD2 "
+            "DOUBLE NOT NULL) NOT NULL)"
+        ).collect()
+
+        table = structured_type_session.table(table_name)
+        non_null_table = structured_type_session.table(non_null_table_name)
+
+        assert table.schema == expected_schema
+        assert non_null_table.schema == expected_non_null_schema
+
+        # Dataframe.union forces a schema_expression call
+        assert table.union(table).schema == expected_schema
+        # Functions used in schema generation don't respect nested nullability so compare query string instead
+        non_null_union = non_null_table.union(non_null_table)
+        assert non_null_union._plan.schema_query == (
```

Comment:
can you also create a test case for nested array and object? like `to_array(... to_array(...))`

Reply:
Added